classes/image_types_ostree.bbclass

```diff
@@ -114,6 +145,23 @@ IMAGE_CMD_ostree () {
 		ln -sf var/roothome root
 	fi
 
+	# deploy SOTA credentials
+	if [ -n "${SOTA_AUTOPROVISION_CREDENTIALS}" ]; then
+		EXPDATE=`openssl pkcs12 -in ${SOTA_AUTOPROVISION_CREDENTIALS} -password "pass:" -nodes 2>/dev/null | openssl x509 -noout -enddate | cut -f2 -d "="`
+
+		if [ `date +%s` -ge `date -d "${EXPDATE}" +%s` ]; then
+			bberror "Certificate ${SOTA_AUTOPROVISION_CREDENTIALS} has expired on ${EXPDATE}"
+		fi
+
+		mkdir -p var/sota
+		cp ${SOTA_AUTOPROVISION_CREDENTIALS} var/sota/sota_provisioning_credentials.p12
+		if [ -n "${SOTA_AUTOPROVISION_URL_FILE}" ]; then
+			export SOTA_AUTOPROVISION_URL="https://`cat ${SOTA_AUTOPROVISION_URL_FILE}`:9000"
```

Comment:
I'm not sure that this is the right port in all cases.  I've send an email around to check.